components/external_storage/export/src/export.rs

```diff
@@ -324,16 +326,25 @@ impl ExternalStorage for EncryptedExternalStorage {
     fn read(&self, name: &str) -> Box<dyn AsyncRead + Unpin + '_> {
         self.storage.read(name)
     }
+    fn read_part(&self, name: &str, off: u64, len: u64) -> Box<dyn AsyncRead + Unpin + '_> {
+        self.storage.read_part(name, off, len)
+    }
     fn restore(
         &self,
         storage_name: &str,
         restore_name: std::path::PathBuf,
+        expected_offset: Option<(u64, u64)>,
         expected_length: u64,
         expected_sha256: Option<Vec<u8>>,
         speed_limiter: &Limiter,
         file_crypter: Option<FileEncryptionInfo>,
     ) -> io::Result<()> {
-        let reader = self.read(storage_name);
+        let reader = if let Some((off, len)) = expected_offset {
+            let r = self.read_part(storage_name, off, len);
+            Box::new(ZstdDecoder::new(BufReader::new(r)))
```

Comment:
Oh, I gave the variant a confusing name.
Maybe It should be called `expected_compressed_range`, because it isn't `None` only when `MetaKV.compress_length` is set.
https://github.com/tikv/tikv/pull/13233/files/af75a6dc60ac3fca8838c11564412103b0bcd959#diff-394395bf71938b7f8728a7689aa8b8c6215a417d04b2ce92a20a08caf2dae19aR326

Reply:
The PR from next week will introduce the field `compression_type` into `DataFileInfo`'s metadata. So `compress+get_range` is triggered  by `compress_length` for time being.